apps/meteor/client/views/room/providers/RoomProvider.tsx

```diff
@@ -102,7 +102,7 @@ const RoomProvider = ({ rid, children }: RoomProviderProps): ReactElement => {
 	const api = useMemo(() => ({}), []);
 
 	if (!pseudoRoom) {
-		return roomQuery.isSuccess ? <RoomNotFound /> : <RoomSkeleton />;
+		return roomQuery.isSuccess && roomQuery.data === undefined ? <RoomNotFound /> : <RoomSkeleton />;
```

Comment:
Shouldn't you also consider the `null` value? Maybe `!roomQuery.data` to cover all falsy values.

Reply:
Actually the `roomQuery.data`  is returning `null` while is still try fetching the data from `room.findOne`, it will only return `undefined` if there is no room.

I can't just do `!roomQuery.data` here because `!null` is `true` and will show the `RoomNotFound` while the data is not fully checked yet.

Reply:
Got it